apps/web/src/server/applications/common/services/address.service.js

```diff
@@ -0,0 +1,25 @@
+import config from '@pins/applications.web/environment/config.js';
+import { findAddressListByPostcode } from '@planning-inspectorate/address-lookup';
+import path from 'node:path';
+import fs from 'node:fs/promises';
+
+/** @typedef {import('@planning-inspectorate/address-lookup').address} address */
+/** @typedef {import('@planning-inspectorate/address-lookup').ValidationErrors} AddressLookupValidationErrors */
+
+/**
+ * Only lookup addresses when not in development *
+ *
+ * @param {string} postcode
+ * @param {any} locals
+ * @returns {Promise<{addressList: address[]; errors?: AddressLookupValidationErrors}>}
+ */
+export default async function (postcode, locals) {
+	if (config.dummyAddressData) {
+		const dummyAddressDataFile = path.join(process.cwd(), 'dummy_address_data.json');
+		const dummyAddressData = JSON.parse(await fs.readFile(dummyAddressDataFile, 'utf8'));
+		const matchingAddresses = { addressList: dummyAddressData[postcode] };
+		return Promise.resolve(matchingAddresses);
+	} else {
+		return findAddressListByPostcode(postcode, locals);
+	}
+}
```

Comment:
Just refactored to use a guard statement so the unhappy path is at the top out of the way.

Also since nothing is awaited, could the async be removed?

```suggestion
 * @returns {Promise<{addressList: address[]; errors?: AddressLookupValidationErrors}>}
 */
export default function (postcode, locals) {
	if (!config.dummyAddressData) {
		return findAddressListByPostcode(postcode, locals);
	}

	const dummyAddressDataFile = path.join(process.cwd(), 'dummy_address_data.json');
	const dummyAddressData = JSON.parse(await fs.readFile(dummyAddressDataFile, 'utf8'));
	const matchingAddresses = { addressList: dummyAddressData[postcode] };
	return matchingAddresses;
}
```

Reply:
Change made as suggested except for the async as this is required due to the await on line 22 when reading the json file.